src/components/Button/ButtonAddPostTopic.js

```diff
@@ -42,12 +51,12 @@ const styles = StyleSheet.create({
     zIndex: 99,
     backgroundColor: COLORS.black
   },
-  buttonContainer: {
-    backgroundColor: COLORS.signed_primary,
+  buttonContainer: (followType) => ({
+    backgroundColor: followType === 'incognito' ? COLORS.anon_secondary : COLORS.signed_primary,
     // height: dimen.size.TOPIC_FEED_POST_BUTTON_HEIGHT,
     borderRadius: dimen.size.FEED_ACTION_BUTTON_RADIUS,
     justifyContent: 'center'
-  },
+  }),
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good as they allow the button's background color to be dynamic based on the `followType` prop. However, it would be better to define the colors used in a constant or configuration file to avoid hardcoding them directly in the component. This will make it easier to manage and update these colors in the future.
```diff
- backgroundColor: followType === 'incognito' ? COLORS.anon_secondary : COLORS.signed_primary,
+ const FOLLOW_TYPE_COLORS = {
+   incognito: COLORS.anon_secondary,
+   default: COLORS.signed_primary,
+ };
+ backgroundColor: FOLLOW_TYPE_COLORS[followType] || FOLLOW_TYPE_COLORS.default,
```
This way, if more follow types are added in the future, you can simply update the `FOLLOW_TYPE_COLORS` object without having to modify the component itself.


<!-- This is an auto-generated comment by OSS CodeRabbit -->